packages/react-components/react-swatch-picker-preview/docs/Spec.md

```diff
@@ -2,62 +2,395 @@
+| Image cell                                                   |                     | ImageSwatch  | ❌        |
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```jsx
+<SwatchPicker aria-label="Font color" layout="grid" columnCount={3}>
+  <ColorSwatch value="#FF1921" aria-label="Red" />
+  <ColorSwatch value="#FFC12E" aria-label="Orange" />
+  <ColorSwatch value="#FEFF37" aria-label="Yellow" />
+  <ColorSwatch value="#00B053" aria-label="Green" />
+  <ColorSwatch value="#00AFED" aria-label="Light Blue" />
+  <ColorSwatch value="#006EBD" aria-label="Blue" />
+  <ColorSwatch value="#712F9E" aria-label="Purple" icon={<SomeIcon />} />
+  <ImageSwatch value="./path/image1.png" disabled aria-label="Space craft" />
+  <ImageSwatch value="./path/image2.png" aria-label="Planets" />
+</SwatchPicker>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+### Layout variants
+
+- Grid
+- Row
+
+For the grid layout maximum recommended amount of swatches is 64 - 8x8 grid.
+For the row layout it's 8 swatches.
+
+To use grid layout it should be more than 4 swatches.
+
+### Swatch Variants
+
+- Color
+- Gradient
+- Image
+- Pattern / texture
+
+### Shapes
+
+- `square` (default)
+- `circular`
+- `rounded`
+
+Border radius for rounded shape can be customized via CSS.
+
+### Size
+
+- `extraSmall`: 20px
+- `small`: 24px
+- `medium` (default): 28px
+- `large`: 32px
+
+Custom size can be set by overriding `width` and `height` of the ColorSwatch or ImageSwatch.
+
+### Spacing
+
+- `small`: 2px
+- `medium` (default): 4px
+
+### States
+
+- `rest`
+- `hover`
+- `pressed`
+- `selected`
+- `selected pressed`
+- `selected hover`
+- `focused`
+- `disabled` - should be used with caution. In case if there are too many disabled swatches it's better to hide them.
+- `empty`
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### SwatchPicker
+
+| Property          | Values                                   | Default   | Purpose                                                      |
+| ----------------- | ---------------------------------------- | --------- | ------------------------------------------------------------ |
+| disabled          | boolean                                  | false     | Whether SwatchPicker is disabled                             |
+| layout            | `row`, `grid`                            | `row`     | Sets layout of the SwatchPicker                              |
+| onColorPreview    | function                                 | undefined | Callback called when color is previewed (on hover or focus). |
+| onColorSelect     | function                                 | undefined | Callback called when color is selected                       |
```

Comment:
wondering what's the callback signature would look like ( not sure if such a detail is needed for Spec though )

we have new `EventHandler` type that should be used https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/convergence/event-handlers-event-type.md#implementation.